pinot-core/src/main/java/org/apache/pinot/core/plan/TransformPlanNode.java

```diff
@@ -43,76 +39,70 @@
 
   private final String _segmentName;
   private final ProjectionPlanNode _projectionPlanNode;
-  private final Set<String> _projectionColumns = new HashSet<>();
-  private final Set<TransformExpressionTree> _expressions = new HashSet<>();
+  private final Set<TransformExpressionTree> _expressions;
   private int _maxDocPerNextCall = DocIdSetPlanNode.MAX_DOC_PER_CALL;
 
-  public TransformPlanNode(IndexSegment indexSegment, BrokerRequest brokerRequest) {
+  public TransformPlanNode(IndexSegment indexSegment, BrokerRequest brokerRequest,
+      Set<TransformExpressionTree> expressionsToPlan) {
```

Comment:
(nit) `expressionsToPlan` -> `expressions`?

Also pass `maxDocsPerBlock` from upper level?

Reply:
`expressions` was too generic and being used in too many places for different purposes, so I thought to call it `expressionsToPlan`.
Replied on `maxDocsPerBlock` above.